src/coreclr/src/jit/codegencommon.cpp

```diff
@@ -5478,13 +5456,8 @@ void CodeGen::genFreeLclFrame(unsigned frameSize, /* IN OUT */ bool* pUnwindStar
     }
```

Comment:
The logic in this if-else block looks similar to `CodeGen::genInstrWithConstant` in codegenarm.cpp
I wonder whether it is worth to extend the latter function in the same way as it is done in codegenarm64.cpp (i.e. add support of reporting unwind information) so it can be used here

Reply:
I'm not sure I see enough similarity to make it useful to combine. This is using a 2-register form, and the other apparently currently doesn't need to deal with modifying SP, but perhaps I'm missing something.

Reply:
Correct, `genInstrWithConstant` is a 2-register instruction that is called by `CodeGen::genStackPointerAdjustment` as `genInstrWithConstant(add, sp, sp, spDelta)`. 

`genStackPointerAdjustment` can be used in `genFreeLclFrame` if `genInstrWithConstant` is extended such that it accepts `bool inUnwindRegion` parameter (as it is done on arm64) and reports unwind information (so `genStackPointerAdjustment` can also report it).

Reply:
That can be done and it will look better, but I do not think that it should be a part of this PR.